components/locid/src/locale.rs

```diff
@@ -58,22 +58,16 @@ use std::str::FromStr;
 /// let loc: Locale = "eN_latn_Us-Valencia_u-hC-H12".parse()
 ///     .expect("Failed to parse.");
 ///
-/// assert_eq!(loc.language, "en");
-/// assert_eq!(loc.script, Some("Latn".parse().unwrap()));
-/// assert_eq!(loc.region, Some("US".parse().unwrap()));
-/// assert_eq!(loc.variants.get(0).unwrap(), "valencia");
+/// assert_eq!(loc.id.language, "en");
+/// assert_eq!(loc.id.script, Some("Latn".parse().unwrap()));
+/// assert_eq!(loc.id.region, Some("US".parse().unwrap()));
+/// assert_eq!(loc.id.variants.get(0).unwrap(), "valencia");
 /// ```
 /// [`Unicode Locale Identifier`]: https://unicode.org/reports/tr35/tr35.html#Unicode_locale_identifier
 #[derive(Default, PartialEq, Eq, Clone, Hash, PartialOrd, Ord)]
 pub struct Locale {
-    /// Language subtag of the Locale
-    pub language: subtags::Language,
-    /// Script subtag of the Locale
-    pub script: Option<subtags::Script>,
-    /// Region subtag of the Locale
-    pub region: Option<subtags::Region>,
-    /// Variant subtags of the Locale
-    pub variants: subtags::Variants,
+    // Language component of the Locale
+    pub id: LanguageIdentifier,
     // Unicode Locale Extensions
     pub extensions: extensions::Extensions,
```

Comment:
Consider pulling the extensions into the Locale struct directly, so that Locale is a set of {langid, unicode, transform, private-use}?

Reply:
I suggested in-place for perf. In most perf-critical scenarios maximization is often doing nothing or modifying a single field. Having to create a whole new LI and copy each field just to add `Script` seems suboptimal.
And in most cases I've experienced you are discaring the non-maximized after maximizing it, so in particular in-place seems like the right choice.

And in the rare case where you want to keep it, you can explicitly copy it and then maximize.

Do you think it's a bad design?

Reply:
@zbraniecki wdyt?

Reply:
Hmm, torn a bit. I can see the appeal, but I also see the appeal of being able to do `is locale.extensions.is_empty()`.

@sffc is your preference strong here?

Reply:
On the other hand, `locale.unicode.keywords.get(...)` is nicer than `locale.extensions.keywords.get(...)`.

> @sffc is your preference strong here?

No, maybe 2/5

Reply:
I think worth considering as a follow up then, but not something to do as part of this PR.